packages/twoside/init.lua

```diff
@@ -72,6 +72,9 @@ function package:_init (options)
    end)
    self.class:registerHook("newpage", spreadHook)
    self.class:registerHook("newpage", switchPage)
+   if not SILE.scratch.headers then
+      SILE.scratch.headers = {}
+   end
```

Comment:
The only place "twoside" uses headers scratch are in `\open-spread` -- but is it approached as it should, there?

Is even this command `\open-spread` in the right package/module? Is the solution general? (The answer is obviously "no" to me, so I'd like seeing good rationales for it...)

Reply:
> Is even this command \open-spread in the right package/module?

I have no idea... Where do you think it should be?

Reply:
> I have no idea... Where do you think it should be?

My deep feeling is that only the **class** knows how to handle page breaks and what do to with "empty" pages, (based on the reason why it it introducing them with so-called spreads).

It's hard to explain with the "limited" book class provided by SILE, but I can try, based on my experience designing books with my own resilient.book class (again, it does have some flaws I am aware of, however)[^1]:

 - Consider a `\part` (yep, resilient has these) followed by a chapter, and both open on odd pages, so a blank even page is gonna be inserted in between. Are you in a "novel" or in a "technical documentation"? Why I am asking? 'Cause in a novel, usually, the book name is on even pages (so it's not a problem to repeat it, at least), but in a technical documentation, the chapter name is usually on even pages (so it has to be cleared after a part...)
 - Consider a `\chapter` followed by another chapter, also opening on an odd page[^2], and a blank even page might be needed. Same question, but trickier: to which chapter belongs the empty page? Should it have a header (not to say a footer), and which one? There's no simple answer, except that "it depends" on some publisher preferences...

And here we're indeed discussing parts, chapters, etc. -- that is, things that are (currently) a class-design topic.

[^1]: If you read its doc, you'll see for instance that resilient uses its own header logic, and that it has provision for "novel" and "technical" headers... Though the underlying implementation is a bit of a hack.
[^2]: Though that ought to be configurable: Some books _do_ allow chapters to begin on even pages... No wonder it took me years to design my resilient.book solution, as all styling decisions are configurable -- and suddenly, all sort of edge cases pop up :)

Reply:
Thanks for your explanation, I got your point.

Well, I did this PR without ever thinking about `twoside` not being the most suitable place for `open-spread` or that scratch is about to deprecate... For me it was way simpler.